js/reference.js

```diff
@@ -1516,6 +1516,30 @@ var ERMrest = (function(module) {
         },
 
         /**
+         * This function takes in a tuple and generates a reference that is
+         * constrained based on the domain_filter_pattern annotation. If this
+         * annotation doesn't exist, it returns this (reference)
+         * `this` is the same as column.reference
+         * @param {ERMrest.ReferenceColumn} column - column that `this` is based on
+         * @param {Object} data - tuple data with potential constraints
+         * @returns {ERMrest.Reference} the constrained reference
+         */
+        filteredRef: function(column, data) {
+            var filteredRef,
+                uri = this.uri;
+
+            if (column.foreignKey.annotations.contains(module._annotations.FOREIGN_KEY)){
+                var filterPattern = column.foreignKey.annotations.get(module._annotations.FOREIGN_KEY).content.domain_filter_pattern;
+                var uriFilter = module._renderTemplate(filterPattern, data);
+                // NOTE: should we check for (uriFilter.trim() !== '') ?
```

Comment:
I don't think `module._renderTemplate` can return anything but the defined string from the filter pattern with the variables replaced with values or `null`. But I wanted to leave that comment there and find out because I wasn't sure.